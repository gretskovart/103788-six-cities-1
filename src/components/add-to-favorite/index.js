import AddToFavorite from './add-to-favorite.tsx';

export default AddToFavorite;
