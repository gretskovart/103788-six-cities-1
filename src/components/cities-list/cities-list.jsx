import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {actionCreators} from '../../reducer/data/data';

import CitiesItem from './../cities-item';
import withActiveItem from './../../hocs/with-active-item';

const CitiesList = (props) => {
  const {onClick, activeItem, changeCity, citiesList} = props;

  return citiesList.map((it) => {
    const _onClick = (evt) => {
      evt.preventDefault();
      onClick(it);
      changeCity(it);
    };

    return (
      <CitiesItem
        city={it}
        key={`id-` + it}
        onClick={_onClick}
        isActive={activeItem === it}
      />
    );
  });
};

const mapDispatchToProps = (dispatch) => ({
  changeCity: (city) => {
    dispatch(actionCreators.changeCity(city));
  }
});

const mapStateToProps = (state) => {
  return {
    activeItem: state.data.activeCity,
    citiesList: state.data.citiesList
  };
};

export {CitiesList};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(withActiveItem(CitiesList));

CitiesList.propTypes = {
  citiesList: PropTypes.array.isRequired,
  onClick: PropTypes.func.isRequired,
  changeCity: PropTypes.func.isRequired,
  activeItem: PropTypes.string.isRequired
};

