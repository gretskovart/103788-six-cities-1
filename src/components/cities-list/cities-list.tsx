import * as React from 'react';
import {connect} from 'react-redux';
import {actionCreators} from '../../reducer/data/data';
import CitiesItem from '../cities-item';
import withActiveItem from '../../hocs/with-active-item';

interface Props {
  activeItem: string;
  changeCity: (city: string) => void;
  citiesList: string[];
  onClick: (city: string) => void;
}

const CitiesList = (props: Props) => {
  const {onClick, activeItem, changeCity, citiesList} = props;

  return citiesList.map((it) => {
    const _onClick = (evt) => {
      evt.preventDefault();
      onClick(it);
      changeCity(it);
    };

    return (
      <CitiesItem
        city={it}
        key={`id-` + it}
        onClick={_onClick}
        isActive={activeItem === it}
      />
    );
  });
};

const mapStateToProps = (state): {activeItem: string, citiesList: string[]} => {
  return {
    activeItem: state.data.activeCity,
    citiesList: state.data.citiesList
  };
};

const mapDispatchToProps = (dispatch): {changeCity: Function} => ({
  changeCity: (city: string) => {
    dispatch(actionCreators.changeCity(city));
  }
});

export {CitiesList};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(withActiveItem(CitiesList));
