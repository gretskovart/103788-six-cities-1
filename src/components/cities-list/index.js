import CitiesList from './cities-list.jsx';

export default CitiesList;
