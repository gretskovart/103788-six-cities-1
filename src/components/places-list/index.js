import PlacesList from './places-list.jsx';

export default PlacesList;
