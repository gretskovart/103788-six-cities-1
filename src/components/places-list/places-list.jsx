import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {actionCreators} from '../../reducer/data/data';

import PlacesItem from './../places-item';
import withActiveItem from './../../hocs/with-active-item';

export class PlacesList extends PureComponent {
  render() {
    const {offers, onClick, activeItem, selectOffer} = this.props;

    return offers.map((it) => {
      const {id, img, isPremium, price, rating, title, type} = it;

      const _onClick = (evt) => {
        evt.preventDefault();
        onClick(title);
      };

      const _onOfferSelect = () => {
        selectOffer(it);
      };

      return (
        <PlacesItem
          id={id}
          key={id}
          img={img}
          isPremium={isPremium}
          price={price}
          rating={rating}
          title={title}
          type={type}
          onClick={_onClick}
          onOfferSelect={_onOfferSelect}
          isActive={activeItem === title}
        />
      );
    });
  }
}

const mapStateToProps = (state) => {
  return {
    offers: state.data.offers
  };
};

const mapDispatchToProps = (dispatch) => ({
  selectOffer: (offer) => {
    dispatch(actionCreators.selectAppartmentDetail(offer));
  }
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(withActiveItem(PlacesList));


PlacesList.propTypes = {
  offers: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        img: PropTypes.string.isRequired,
        isPremium: PropTypes.bool.isRequired,
        price: PropTypes.number.isRequired,
        rating: PropTypes.number.isRequired,
        title: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
        coordinates: PropTypes.arrayOf(PropTypes.number).isRequired
      })
  ).isRequired,
  onClick: PropTypes.func.isRequired,
  selectOffer: PropTypes.func.isRequired,
  activeItem: PropTypes.string
};
