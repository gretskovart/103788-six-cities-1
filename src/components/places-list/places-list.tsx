import * as React from 'react';
import {PureComponent} from 'react';
import {connect} from 'react-redux';
import {actionCreators} from '../../reducer/data/data';
import PlacesItem from '../places-item';
import withActiveItem from '../../hocs/with-active-item';
import {types} from '../../helpers';

interface Props {
  offers: types.OfferType[];
  onClick: Function;
  activeItem?: string;
  selectOffer: Function;
  placesType: string;
  selectActiveOffer: Function;
}

class PlacesList extends PureComponent<Props> {
  constructor(props) {
    super(props);
  }

  render() {
    const {offers, onClick, activeItem, selectOffer, placesType, selectActiveOffer} = this.props;

    return offers.map((it) => {
      const {id, img, isPremium, price, rating, title, type, isFavorite} = it;

      const _onClick = (evt) => {
        evt.preventDefault();

        onClick(id);
        selectActiveOffer(id);
      };

      const _onOfferSelect = () => {
        selectOffer(it);
      };

      return (
        <PlacesItem
          id={id}
          key={id}
          img={img}
          isPremium={isPremium}
          price={price}
          rating={rating}
          title={title}
          type={type}
          onClick={_onClick}
          onOfferSelect={_onOfferSelect}
          isActive={activeItem === id}
          placesType={placesType}
          isFavorite={isFavorite}
        />
      );
    });
  }
}

const mapStateToProps = (state) => {
  return {
    offers: state.data.offers
  };
};

const mapDispatchToProps = (dispatch) => ({
  selectOffer: (offer: types.OfferType) => {
    dispatch(actionCreators.selectAppartmentDetail(offer));
  },
  selectActiveOffer: (item: string) => {
    dispatch(actionCreators.selectActiveOffer(item));
  }
});

const WrappedPlacesList = connect(
    null,
    mapDispatchToProps)(
    withActiveItem(PlacesList));

export default connect(
    mapStateToProps,
    null
)(WrappedPlacesList);

export {WrappedPlacesList, PlacesList};
