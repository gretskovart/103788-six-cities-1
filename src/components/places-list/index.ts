import {WrappedPlacesList, PlacesList} from './places-list';

export {WrappedPlacesList, PlacesList};
