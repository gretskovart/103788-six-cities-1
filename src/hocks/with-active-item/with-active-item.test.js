import React from 'react';
import renderer from 'react-test-renderer';
import {PlacesList} from './../../components/places-list';
import {CitiesList} from './../../components/cities-list';

const mock = {
  city: `Paris`,
  activeItem: `Amsterdam`
};

describe(`withActiveItem renders correctly`, () => {
  const {city, activeItem} = mock;

  describe(`PlacesList`, () => {
    it(`renders component correctly`, () => {
      const tree = renderer.create(<PlacesList
        offers={[]}
        activeItem={city}
        onClick={jest.fn()}
      />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe(`CitiesList`, () => {
    it(`renders component correctly`, () => {
      const tree = renderer.create(<CitiesList
        offers={[]}
        onClick={jest.fn()}
        changeCity={jest.fn()}
        activeItem={activeItem}
      />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
