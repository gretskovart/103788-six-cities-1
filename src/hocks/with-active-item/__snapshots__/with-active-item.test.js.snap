// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withActiveItem renders correctly CitiesList renders component correctly 1`] = `null`;

exports[`withActiveItem renders correctly PlacesList renders component correctly 1`] = `null`;
